src/lib/annotations/doc/DocAnnotator.js

```diff
@@ -315,7 +316,7 @@ class DocAnnotator extends Annotator {
 
         if (!thread && this.notification) {
             this.emit('annotationerror', __('annotations_create_error'));
-        } else if (thread) {
+        } else if (thread && (type !== TYPES.draw || location.page)) {
```

Comment:
It's a little bit messy but a DrawingThread shouldn't be added to the threadmap until it has a page. On DrawingThread creation, the page is undefined so adding it to the threadmap would add it to threadmap[page1] rather than the correct page.

Then on lookup we would be looking for the thread in threadmap[correctPage] where correct page may not be page1.